assets/app/scripts/controllers/edit/buildConfig.js

```diff
@@ -0,0 +1,617 @@
+              var projects = projects.by("metadata.name");
+              for (var name in projects) {
+                $scope.buildFrom.projects.push(name);
+                $scope.pushTo.projects.push(name);
+              }
+
+              // If builder or output image reference kind is DockerImage select the first imageSteam and imageStreamTag
+              // in the picker, so when the user changes the reference kind to ImageStreamTag the picker is filled with
+              // default(first) value.
+              var builderSelectFirstOption = ($scope.options.pickedBuildFromType === "DockerImage") ? true : false;
+              var outputSelectFirstOption = ($scope.options.pickedPushToType === "DockerImage") ? true : false;
+              $scope.updateBuilderImageStreams($scope.options.pickedBuildFromNamespace, builderSelectFirstOption);
+              $scope.updateOutputImageStreams($scope.options.pickedPushToNamespace, outputSelectFirstOption);
+
+              if ($scope.sources.image) {
+                $scope.imageSourceBuildFrom.projects = angular.copy($scope.buildFrom.projects);
+                var imageSourceSelectFirstOption = ($scope.options.pickedImageSourceType === "DockerImage") ? true : false;
+                $scope.updateImageSourceImageStreams($scope.options.pickedImageSourceNamespace, imageSourceSelectFirstOption);
+              }
+
+              angular.element(document).ready(function () {
+                // $('.combobox').combobox();
+              });
+            });
+            $scope.loaded = true;
+            // If we found the item successfully, watch for changes on it
+            watches.push(DataService.watchObject("buildconfigs", $routeParams.buildconfig, context, function(buildConfig, action) {
+              if (action === "DELETED") {
+                $scope.alerts["deleted"] = {
+                  type: "warning",
+                  message: "This build configuration has been deleted."
+                };
+              }
+              $scope.buildConfig = buildConfig;
+            }));
+          },
+          // failure
+          function(e) {
+            $scope.loaded = true;
+            $scope.alerts["load"] = {
+              type: "error",
+              message: "The build configuration details could not be loaded.",
+              details: "Reason: " + $filter('getErrorDetails')(e)
+            };
+          }
+        );
+      }));
+
+    $scope.aceLoaded = function(editor) {
+      var session = editor.getSession();
+      session.setOption('tabSize', 2);
+      session.setOption('useSoftTabs', true);
+      editor.$blockScrolling = Infinity;
+
+      // Resize the editor based on window width.
+      var updateEditorWidth = function() {
+        var width = $('.edit-form .editor').outerHeight();
+        $('.edit-form .dockerfile-editor').animate({
+          height: width + 'px'
+        }, 30, function() {
+          editor.resize();
+        });
+      };
```

Comment:
The function is `updateEditorWidth`, but we seem to be changing the height. Since there are many other fields on the page, we can probably use a fixed height and remove the resize handler altogether.